In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('kc_house_data.csv')

In [3]:
df

id             date     price  bedrooms  bathrooms  \
0      7129300520  20141013T000000  221900.0         3       1.00   
1      6414100192  20141209T000000  538000.0         3       2.25   
2      5631500400  20150225T000000  180000.0         2       1.00   
3      2487200875  20141209T000000  604000.0         4       3.00   
4      1954400510  20150218T000000  510000.0         3       2.00   
...           ...              ...       ...       ...        ...   
21608   263000018  20140521T000000  360000.0         3       2.50   
21609  6600060120  20150223T000000  400000.0         4       2.50   
21610  1523300141  20140623T000000  402101.0         2       0.75   
21611   291310100  20150116T000000  400000.0         3       2.50   
21612  1523300157  20141015T000000  325000.0         2       0.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
0             1180      5650     1.0           0     0  ...      7   
1             2570      7242     2.0           0     0  ...      7   
2              770     10000     1.0           0     0  ...      6   
3             1960      5000     1.0           0     0  ...      7   
4             1680      8080     1.0           0     0  ...      8   
...            ...       ...     ...         ...   ...  ...    ...   
21608         1530      1131     3.0           0     0  ...      8   
21609         2310      5813     2.0           0     0  ...      8   
21610         1020      1350     2.0           0     0  ...      7   
21611         1600      2388     2.0           0     0  ...      8   
21612         1020      1076     2.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
0            1180              0      1955             0    98178  47.5112   
1            2170            400      1951          1991    98125  47.7210   
2             770              0      1933             0    98028  47.7379   
3            1050            910      1965             0    98136  47.5208   
4            1680              0      1987             0    98074  47.6168   
...           ...            ...       ...           ...      ...      ...   
21608        1530              0      2009             0    98103  47.6993   
21609        2310              0      2014             0    98146  47.5107   
21610        1020              0      2009             0    98144  47.5944   
21611        1600              0      2004             0    98027  47.5345   
21612        1020              0      2008             0    98144  47.5941   

          long  sqft_living15  sqft_lot15  
0     -122.257           1340        5650  
1     -122.319           1690        7639  
2     -122.233           2720        8062  
3     -122.393           1360        5000  
4     -122.045           1800        7503  
...        ...            ...         ...  
21608 -122.346           1530        1509  
21609 -122.362           1830        7200  
21610 -122.299           1020        2007  
21611 -122.069           1410        1287  
21612 -122.299           1020        1357  

[21613 rows x 21 columns]

houses built after year 2000

In [5]:
adf = df[df['yr_built']>=2000]

In [6]:
adf

id             date      price  bedrooms  bathrooms  \
5      7237550310  20140512T000000  1230000.0         4       4.50   
9      3793500160  20150312T000000   323000.0         3       2.50   
29     1873100390  20150302T000000   719000.0         4       2.50   
30     8562750320  20141110T000000   580500.0         3       2.50   
31     2426039314  20141201T000000   280000.0         2       1.50   
...           ...              ...        ...       ...        ...   
21608   263000018  20140521T000000   360000.0         3       2.50   
21609  6600060120  20150223T000000   400000.0         4       2.50   
21610  1523300141  20140623T000000   402101.0         2       0.75   
21611   291310100  20150116T000000   400000.0         3       2.50   
21612  1523300157  20141015T000000   325000.0         2       0.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
5             5420    101930     1.0           0     0  ...     11   
9             1890      6560     2.0           0     0  ...      7   
29            2570      7173     2.0           0     0  ...      8   
30            2320      3980     2.0           0     0  ...      8   
31            1190      1265     3.0           0     0  ...      7   
...            ...       ...     ...         ...   ...  ...    ...   
21608         1530      1131     3.0           0     0  ...      8   
21609         2310      5813     2.0           0     0  ...      8   
21610         1020      1350     2.0           0     0  ...      7   
21611         1600      2388     2.0           0     0  ...      8   
21612         1020      1076     2.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
5            3890           1530      2001             0    98053  47.6561   
9            1890              0      2003             0    98038  47.3684   
29           2570              0      2005             0    98052  47.7073   
30           2320              0      2003             0    98027  47.5391   
31           1190              0      2005             0    98133  47.7274   
...           ...            ...       ...           ...      ...      ...   
21608        1530              0      2009             0    98103  47.6993   
21609        2310              0      2014             0    98146  47.5107   
21610        1020              0      2009             0    98144  47.5944   
21611        1600              0      2004             0    98027  47.5345   
21612        1020              0      2008             0    98144  47.5941   

          long  sqft_living15  sqft_lot15  
5     -122.005           4760      101930  
9     -122.031           2390        7570  
29    -122.110           2630        6026  
30    -122.070           2580        3980  
31    -122.357           1390        1756  
...        ...            ...         ...  
21608 -122.346           1530        1509  
21609 -122.362           1830        7200  
21610 -122.299           1020        2007  
21611 -122.069           1410        1287  
21612 -122.299           1020        1357  

[4759 rows x 21 columns]

2. most expensive houses by year

In [30]:
mxHouses = adf.groupby('yr_built')['price'].idxmax()
mxHouses = adf.loc[mxHouses]
mxHouses = mxHouses[['yr_built','price']]
mxHouses

yr_built      price
16302      2000  3710000.0
9254       2001  6890000.0
300        2002  3080000.0
12370      2003  4210000.0
2626       2004  4500000.0
19148      2005  3640000.0
16969      2006  3100000.0
18482      2007  3650000.0
4149       2008  4000000.0
1448       2009  5350000.0
20154      2010  2950000.0
7432       2011  2580000.0
10468      2012  2000000.0
21530      2013  3000000.0
5880       2014  3420000.0
21333      2015  2000000.0

3. houses with most bedrooms

In [28]:
mbdr = adf.groupby('yr_built')['bedrooms'].idxmax()
mbdr = adf.loc[mbdr]
mbdr = mbdr[['yr_built','bedrooms','price']]
mbdr

yr_built  bedrooms      price
2668       2000         6  1380000.0
12017      2001         7   936000.0
1539       2002         6  1300000.0
2852       2003         6   635000.0
18975      2004         7  1240000.0
9840       2005         6  1140000.0
20393      2006         6   300000.0
2345       2007         6   495000.0
13314      2008        10  1150000.0
20984      2009         6   343000.0
1877       2010         5   820000.0
2694       2011         5   830000.0
19793      2012         6   729999.0
8120       2013         7   450000.0
1882       2014         6  1130000.0
7526       2015         5   614285.0

4. most bathrooms

In [26]:
mbth = adf.groupby('yr_built')['bathrooms'].idxmax()
mbth = df.loc[mbth]
mbth = mbth[['yr_built','bathrooms','price']]
mbth

yr_built  bathrooms      price
6501       2000       5.75  1940000.0
9254       2001       7.75  6890000.0
300        2002       5.00  3080000.0
12370      2003       6.00  4210000.0
16773      2004       5.75  1920000.0
19148      2005       6.00  3640000.0
3100       2006       5.50  1500000.0
20578      2007       6.50  1180000.0
4149       2008       5.50  4000000.0
1448       2009       5.00  5350000.0
21506      2010       6.50  2240000.0
19770      2011       4.00   465000.0
6389       2012       4.50   600000.0
21148      2013       5.00  1800000.0
5880       2014       5.00  3420000.0
20251      2015       4.50   771005.0

5. max floors

In [27]:
mflrs = adf.groupby('yr_built')['floors'].idxmax()
mflrs = adf.loc[mflrs]
mflrs = mflrs[['yr_built','price','floors']]
mflrs

yr_built      price  floors
438        2000   438000.0     3.0
406        2001   450000.0     3.0
667        2002   655000.0     3.0
985        2003   535000.0     3.0
663        2004   312000.0     3.0
10077      2005   435000.0     3.5
14885      2006   525000.0     3.5
20308      2007   525000.0     3.5
15424      2008   479000.0     3.5
2478       2009   275000.0     3.0
116        2010   518500.0     3.0
3452       2011  1230000.0     3.0
4346       2012   715000.0     3.0
1931       2013   598800.0     3.0
63         2014   549000.0     3.0
643        2015   729500.0     3.0

6 . houses with an area larger than the mean area of all houses

In [36]:
sqftLivingM = adf['sqft_living'].mean()
sqftLivingM = adf[adf['sqft_living']>sqftLivingM]
sqftLivingM = sqftLivingM[['sqft_living','price']]
sqftLivingM.head(10)

sqft_living      price
5           5420  1230000.0
29          2570   719000.0
40          2570   625000.0
42          3595   861990.0
45          3160   488000.0
79          3450   720000.0
87          2770   461000.0
91          3520  1100000.0
114         3230   480000.0
153         5180  2250000.0

houses with area larger than the mean of the sqft_lot

In [37]:
sqftLotM = adf['sqft_lot'].mean()
sqftLotM = adf[adf['sqft_lot']>sqftLotM]
sqftLotM = sqftLotM[['sqft_lot','price']]
sqftLotM.head(10)

sqft_lot      price
5      101930  1230000.0
45      13603   488000.0
79      39683   720000.0
114     16171   480000.0
147     60984   430000.0
153     19850  2250000.0
162     16603   900000.0
213     17936   465000.0
277     26300   770000.0
300     18641  3080000.0

conclusion:
by analyzing the first dataset about most expensive houses by year we conclude that yr_built doesn't affect the highest price because as we can see in year 2001, the hightest price of a house was 6,890,000 . while the most expensive house built in year 2004 was 4,5000,000.
moving along to houses with most bedrooms in every year from 2000 to 2015 we have:
houses with higher number of bedrooms tend to be more expensive than houses with  lower number of bedrooms in most cases
also with bedrooms we can see a similar pattern, as number of bathrooms increases in most cases, the price of the houses also increases.
next, by comparing the houses with highest number of floors each year with their price we get the following,
for every year the number of max floors is relatively the same, only in 4 cases we found that the number is higher (3.5 floors) from year 2005 till 2008. while the rest of the years the highest number of floors is (3 floors). so there's no significant change in number of floors with price.
while analyzing a sample of 10 houses with an area larger than the mean area of all houses we conclude that as an area increases, the price also increases.
however, it's not the same case while comparing the top 10 houses with an area of lot greater than the mean of houses. we can see a variety of area with a great variety of prices


IN a nutshell:
while factors like number of bedrroms, bathrooms and area affect the house prices positively, the year of construction and number of floors don't consistently impact the highest price of houses

fucntion whcih takes in a set of coordinates(lat and long) and another set, and calculates the distance between them

The .at and iat methods are used to access specific elemnts of a DataFrame by label or integer position. 
They are similar to loc and iloc but provide a more efficient way to access individual elements and give and increase in performance